sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -137,30 +137,40 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
    */
   struct TransportOptions final
   {
-#if !defined(BUILD_TRANSPORT_CUSTOM_ADAPTER)
-#if defined(BUILD_TRANSPORT_WINHTTP_ADAPTER) || defined(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
     /**
```

Comment:
I'm so happy to see these gone! If you want, I can approve the PR now, but you still need to update two files for `wil` dependency, and let's follow up on the 3rd party naming (tomorrow, I think) - we can do naming update as a follow up PR if we come to a different conclusion.

Reply:
If you're keeping the template solution, the naming style discussion is probably not important for this PR, but we still can discuss it, independently, for the future.

Reply:
`UniqueHandle` is also a great name, I think! Unique pointer, unique handle.

Reply:
I'm keeping the template solution - I'm going to need it for uAMQP in the near future as well, so this was something I had to do eventually.

PS: I'm not sure what the changes I should make to the config.cmake.in file.

Reply:
Here, just put "wil" in the same manner as we do openssl: https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/vcpkg/Config.cmake.in#L13-L15

Reply:
Makes sense.